scripts/lib_oval.py

```diff
@@ -150,8 +150,14 @@ def __init__(self, tree):
             gen.setSchemaVersion("5.10.1")
             root.append(gen.get_element())
             return        
-        
+
         self.tree = tree
+        self.id_to_definition = { el.getId(): el for el in self.getDefinitions()} if self.getDefinitions() else {}
```

Comment:
I'd recommend:
- because this cache is really only used by getElementByID, I suggest moving this into that method, adding code to getElementByID (e.g. `if not hasattr(self, 'id_to_test'): self.id_to_test = ...`
- because oval ids must be unique in this scope, simplify by changing cache to 1 dictionary (oval_id:element) with all element types
- then you can simplify getElementByID, removing all of the oval_type related code

Reply:
Yeah that is  better, I will make the changes.

Reply:
Thank you! This code doesn't get a lot of attention and so long as we're all looking at it, it's terrific if we can simplify it a bit along with your performance improvements!

Reply:
Sure , it has a (big) room for improvement , by the way how are the tests supposed to be run on this one? I didn't find any documentation regarding it.